3d/vector_types.ksy

```diff
@@ -0,0 +1,54 @@
+meta:
```

Comment:
1. It may make sense to move it into `/common/`
2. I guess each type name should contain the types of elements.
3. I wonder if we should embed a 2d vector into 3d one.
4.
```yaml
-affected-by:
  - 135  # would allow to drop the types with hardcoded types of fields
  - 65  # implementation of constant-size types via raw structs would make embedding one vec into another one more efficient (effectively zero-cost for some languages) 
`

https://github.com/kaitai-io/kaitai_struct/issues/135 https://github.com/kaitai-io/kaitai_struct/issues/65

Reply:
Embedding a vector so that you can access `.xy` more efficiently is probably no good either because it assumes that you prefer the X and Y coords over Z and that is not always the case because it really depends on which way the object is oriented. In graphics programming I have seen things like `.zx` or even things like `.xyxx`.

So really all I’d want here is straight call-through methods which avoid caching so that I can do those swizzles.

Reply:
As for moving to common, yeah, I worried about moving it to common _exactly because_ then I’d have to deal with people commenting about the types of elements, or even the endianness, or other things along those lines.

Reply:
What would _really_ be convenient here is parametrised imports so that I can import vector types here and pass in the element type and endianness.